src/implementations/utils/wallet.ts

```diff
@@ -21,33 +36,51 @@ export const getPrivateKey = ({ keyFile, key }: WalletOption): string | undefine
   }
   return key || getKeyFromFile(keyFile) || process.env["OA_PRIVATE_KEY"];
 };
-export const getWallet = async ({
-  keyFile,
-  key,
+
+export const getWalletOrSigner = async ({
   network,
-  encryptedWalletPath,
   progress = defaultProgress("Decrypting Wallet"),
-}: WalletOption & Partial<NetworkOption> & { progress?: (progress: number) => void }): Promise<Wallet> => {
+  ...options
+}: WalletOrSignerOption & Partial<NetworkOption> & { progress?: (progress: number) => void }): Promise<
+  Wallet | ConnectedSigner
+> => {
   const provider =
     network === "local"
       ? new providers.JsonRpcProvider()
       : getDefaultProvider(network === "mainnet" ? "homestead" : network); // homestead => aka mainnet
-  if (encryptedWalletPath) {
+  if (isWalletOption(options)) {
     const { password } = await inquirer.prompt({ type: "password", name: "password", message: "Wallet password" });
 
-    const file = await readFile(encryptedWalletPath);
+    const file = await readFile(options.encryptedWalletPath);
     const wallet = await ethers.Wallet.fromEncryptedJson(file, password, progress);
     signale.info("Wallet successfully decrypted");
     return wallet.connect(provider);
-  } else {
-    const privateKey = getPrivateKey({ key, keyFile });
+  } else if (isAwsKmsSignerOption(options)) {
+    const { secretAccessKey } = await inquirer.prompt({
+      type: "password",
+      name: "secretAccessKey",
+      message: "Secret access key",
+    });
+    const kmsCredentials = {
+      accessKeyId: options.accessKeyId, // credentials for your IAM user with KMS access
+      secretAccessKey, // credentials for your IAM user with KMS access
+      region: options.region,
+      keyId: options.kmsKeyId,
+    };
 
-    if (!privateKey)
-      throw new Error(
-        "No private key found in OA_PRIVATE_KEY, key, key-file, please supply at least one or supply an encrypted wallet path"
-      );
+    const provider = ethers.providers.getDefaultProvider("ropsten");
```

Comment:
is the awskms only open for ropsten for now ?

Reply:
fuck

Reply:
I forgot to remove this line